### PR TITLE
Implement MustEncode

### PR DESCRIPTION
**Pull request by monkrus**
## Changes

Implement MustEncode for messages and block components. Files affected: block.go header.go
-
-
-

## Tests

```

```

## Checklist

- [ x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [x] All integration tests and required coverage checks are passing

## Issues

-

### CODE REVIEW

**Comment by monkrus** [comment]
Please review when possible.

File: dot/types/block.go [hunk 1]
```diff
@@ -65,6 +65,15 @@ func (b *Block) Encode() ([]byte, error) {
 	return append(enc, encBody...), nil
 }
 
+// MustEncode verifies that block is encoded
```
Comment (noot) [inline dot/types/block.go:68]: could you update this to say "MustEncode returns the SCALE encoded block and panics if it fails to encode" or something along those lines? same with for header

**Comment by monkrus** [comment]
I apologize for the long wait, but that is the best I could come up with due to the skills/time restrictions. I hate to leave a comment like that, but if someone has time to finish writing those test cases, I would appreciate that a lot.

**Comment by noot** [comment]
@monkrus no worries! overall, it looks good to me. I'll take a look today and try to get this merged in :)

**Pull request closed without merging**